### PR TITLE
docs: Fix spelling and formatting

### PR DESCRIPTION
**Pull request by jodh-intel**
Correct typos and formatting issues.

Fixes: #1734.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: virtcontainers/experimental/README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Experimental package description
 
+* [1. What are "experimental" features?](#1-what-are-experimental-features)
```
Comment (grahamwhaley) [inline virtcontainers/experimental/README.md]: Heh, bullets in titles - not sure we do that anywhere else, and I can just see an update nightmare coming down the road. Would you consider dropping those whilst you are here?

Comment (jodh-intel) [reply virtcontainers/experimental/README.md]: Agreed - done.

**Comment by jodh-intel** [comment]
Doc-only PR, so merging...

**Pull request merged**